haystack/modeling/training/base.py

```diff
@@ -252,7 +252,7 @@ def train(self):
                 vocab_size2=len(self.data_silo.processor.passage_tokenizer),
             )
         elif (
-            self.model.language_model.name != "debertav2"
+            self.model.language_model.name.lower() != "debertav2"
```

Comment:
The correct lowercase form is `deberta-v2` as far as I know. Is there cases in which DeBERTa is lower cased this way? Does this fix actually solve the bug in training you observed?

During the language models refactoring this lower case conversions were a big headache. Can you provide examples? Is this is the correct lower case form, there are other code paths that need to be fixed.

Reply:
Yes this does solve the bug in training since the model name I get is this
```python
print(self.model.language_model.name)
"DebertaV2"
```

when loading this reader
```python
reader = FARMReader(model_name_or_path="microsoft/mdeberta-v3-base", use_gpu=True, use_confidence_scores=False)
```

so I'm not sure that the correct lowercase form is `deberta-v2` in this case. Unless there is a bug somewhere else.

Reply:
Ok, I'd rather invert the fix to avoid confusion.

```suggestion
            self.model.language_model.name != "DebertaV2"
```